src/Couchbase.Lite.Tests.Shared/ReplicationTest.cs

```diff
@@ -1291,6 +1319,107 @@ public void TestConflictResolverExceptionsNoneMatchDBThrown()
             }
         }
 
+        [Fact]
+        public void TestConflictResolverReturningBlob()
+        {
+            var returnRemoteDoc = true;
+            TestConflictResolverWinsInBlobs(returnRemoteDoc);
+            TestConflictResolverWinsInBlobs(!returnRemoteDoc);
+
+            //return new doc with a blob object
+            CreateReplicationConflictInBlobs();
+
+            var config = CreateConfig(false, true, false);
+
+            config.ConflictResolver = new TestConflictResolver((conflict) =>
+            {
+                var evilByteArray = new byte[] { 6, 6, 6 };
+
+                var doc = new MutableDocument();
+                doc.SetBlob("blob", new Blob("text/plaintext", evilByteArray));
+                return doc;
+            });
+
+            RunReplication(config, 0, 0);
+
+            using (var doc = Db.GetDocument("doc1")) {
+                doc.GetBlob("blob")?.Content.Should().ContainInOrder(new byte[] { 6, 6, 6 });
+            }
+        }
+
+        [Fact]
+        public void TestConflictResolverReturningBlobFromDifferentDB()
+        {
+            var tmpDoc = new MutableDocument("doc1");
+            var evilByteArray = new byte[] { 6, 6, 6 };
+            using (var thirdDb = new Database("different_db")) {
+                tmpDoc.SetBlob("blob", new Blob("text/plaintext", evilByteArray));
+                thirdDb.Save(tmpDoc);
+
+                var differentDbResolver = new TestConflictResolver((conflict) => tmpDoc);
```

Comment:
This isn't quite correct.  You need to make a new document with the blob from another DB.  If it's saved in a different DB then this is the same test as the differing DB test.